agimus_controller/agimus_controller/warm_start_factory.py

```diff
@@ -0,0 +1,17 @@
+from agimus_controller.warm_start_base import WarmStartBase
+
+
+def _create_warm_start_from_previous_solution() -> WarmStartBase:
+    pass
+
+
+def _create_warm_start_from_diffusion_model() -> WarmStartBase:
+    pass
+
+
+def create_warm_start(name: str) -> WarmStartBase:
```

Comment:
it's not completely clear to me how we will use this function, when looking at the skeleton of the mpc, I have the impression that you want to call this function only once at the setup, but when we will use diffusion model, we wil ll also sometimes use previous solution as warm start, so I'm a bit confused, maybe deciding what warm-start we use should be decided in the classes that will inherit from the WarmStartBase class ?

Reply:
So I assumed that warmstart can have an internal state machine and the factory only gives us the implementation we want to plug in to the whole system. The behaviour whether we use gravity compensation or something else only differs when there is no previous solution. At least this is my understanding